### DISCUSSION TITLE
multi module, using wildcards in module names

### DISCUSSION THREAD

**Discussion by Andy1616** [prometheus/snmp_exporter#1064] [Ideas]
Hi all 

with regards to multi modules

A call like
http://localhost:9116/snmp?target=192.0.0.8&module=if_mib&module=arista_sw_1&module=arista_sw&module_2=arista_sw_3
could also be done like that
http://localhost:9116/snmp?target=192.0.0.8&module=if_mib&module=arista_sw_.*

Benefits:
In case you add/remove a module from that "group" nothing needs to be done in prometheus, all only local to exporter.
All tuning in terms or parallelism is exporter only, and decoupled from settings in prometheus

to be discussed what pattern to use if regex style of DB style, I see several ways possible... 

What are your thoughts?
Thanks for feedback

**Comment by SuperQ** [comment]
I would do regex or glob matching. Probably not SQL style.

Seems a bit like over-optimization. If targets support higher walk concurrency, maybe what we should support is in-module concurrency in addition to module-by-module concurrency.

**Comment by Andy1616** [comment]
Hi
"in module concurrency" would address the same - only seems to be a bit harder to implement as multi module already is present...

**Reply by SuperQ** [reply]
It shouldn't be that difficult. We would need a concurrency configuration per module (maybe `get_concurrency` and `walk_concurrency` if we want separate params for those). Then add a similar work pool to [ScrapeTarget](https://github.com/prometheus/snmp_exporter/blob/887c9d6f1b99784647e6af2aff9013484e62445d/collector/collector.go#L212).
